python/ray/air/checkpoint.py

```diff
@@ -381,6 +444,15 @@ def from_checkpoint(cls, other: "Checkpoint") -> "Checkpoint":
             obj_ref=other._obj_ref,
         )
 
+    @classmethod
+    def _check_type_compatability(cls, serialized_type: Type["Checkpoint"]):
+        if not issubclass(serialized_type, cls):
```

Comment:
I think you need to check subclass in both directions. And then load back with the subclass, whether that's `serialized_type` or `cls`.